src/IO/H5/Helpers.hpp

```diff
@@ -113,9 +113,43 @@ bool contains_attribute(hid_t file_id, const std::string& group_name,
 
 /*!
  * \ingroup HDF5Group
- * \brief Read a DataVector from a dataset in a group
+ * \brief Open a dataset in an HDF5 group with name "dataset_name"
  */
-DataVector read_data(hid_t group_id, const std::string& dataset_name);
```

Comment:
Since you're replacing `read_data` you should make a symmetric change with `write_data`

Reply:
What does this mean? To create write_rankX_dataset functions?

Reply:
`write_data` is built sufficiently differently that it doesn't make sense to re-write it at this point. I was able to use it easily in a test in which I created, wrote to, read from (using the new `read_data`), and removed an H5 file.